File: tests/elixir/runner.ex
Code.require_file("tests/read.ex", __DIR__)
Code.require_file("tests/write.ex", __DIR__)
Code.require_file("tests/parse.ex", __DIR__)

file_path = System.get_env("TEST_FILE")
verbose = System.get_env("VERBOSE") == "true"
test_mode = (System.get_env("TEST_MODE") || "full")
test_name = System.get_env("TEST_NAME")
test_type = System.get_env("TEST_TYPE")
block_size = System.get_env("KB_BLOCK_SIZE") || 1

if test_type == "" or test_type == nil do
  raise("You must supply TEST_TYPE")
end

if test_type == "read" do
  IOBenchRead.run(file_path, test_mode, verbose, test_name)
end

if test_type == "parse" do
  IOBenchParse.run(file_path, test_mode, verbose, test_name)
end

if test_type == "write" do
  { block_size, _ } = Integer.parse(block_size)
  IOBenchWrite.run(block_size, file_path, test_mode, verbose, test_name)
end

File: tests/python/runner.py
import os
import tests.base
from tests.read import IOBenchRead
from tests.write import IOBenchWrite
from tests.parse import IOBenchParse

class IOBenchRunner:
  def start(self, env):
    file_path  = env.get('TEST_FILE')
    test_type  = env.get('TEST_TYPE')
    test_mode  = (env.get('TEST_MODE') or 'full').lower()
    test_name  = (env.get('TEST_NAME') or os.path.basename(file_path))
    verbose    = (env.get('VERBOSE') == 'true')
    block_size = (int) (env.get('KB_BLOCK_SIZE') or 1)

    if test_type == None:
      raise(Exception('You must suply TEST_TYPE with one the values: read, write, read+write'))

    if test_type == 'read':
      bench = IOBenchRead(file_path, test_mode, test_name, verbose)
      bench.run()
    elif test_type == 'write':
      bench = IOBenchWrite(block_size, file_path, test_mode, test_name, verbose)
      bench.run()
    elif test_type == 'parse':
      bench = IOBenchParse(file_path, test_mode, test_name, verbose)
      bench.run()


IOBenchRunner().start(os.environ)